src/bellpepper/r1cs.rs

```diff
@@ -43,12 +43,11 @@ impl<G: Group> NovaWitness<G> for SatisfyingAssignment<G> {
     shape: &R1CSShape<G>,
     ck: &CommitmentKey<G>,
   ) -> Result<(R1CSInstance<G>, R1CSWitness<G>), NovaError> {
-    let W = R1CSWitness::<G>::new(shape, self.aux_assignment())?;
-    let X = &self.input_assignment()[1..];
+    let W = R1CSWitness::<G>::new(shape, self.aux_assignment().to_vec())?;
 
     let comm_W = W.commit(ck);
 
-    let instance = R1CSInstance::<G>::new(shape, &comm_W, X)?;
+    let instance = R1CSInstance::<G>::new(shape, comm_W, self.input_assignment().to_vec())?;
```

Comment:
Since two people have asked me about why there are still `clone`s here, I want to clarify. This function, `r1cs_instance_and_witness`, is no longer being called in `prove_step`. This change is purely atheistic -- it moves the `.to_vec()` call inside `R1CSWitness::new`/`R1CSInstance::new` to be explicit on construction -- so the compiler is happy. 

We do not call this function, there are no extra clones.